### PR TITLE
Buck build configuration and scripts

### PR DESCRIPTION
**Pull request by maxme**
To test this PR, you need to:

* Install Buck https://facebook.github.io/buck/setup/install.html
* Bootstrap dependencies `$ ./tools/fetch_buck_dependencies.py extlibs`
* Then build `$ buck build wpandroid` or install and run: `$ buck install --run wpandroid`

### CODE REVIEW

**Comment by maxme** [comment]
I forgot to parse the `local.properties` file here: https://github.com/wordpress-mobile/WordPress-Android/blob/feature/buck-build-tool-take3/tools/fetch_buck_dependencies.py#L10

**Comment by maxme** [comment]
> I forgot to parse the local.properties file here: https://github.com/wordpress-mobile/WordPress-Android/blob/feature/buck-build-tool-take3/tools/fetch_buck_dependencies.py#L10

fixed in 0745f7d

**Comment by roundhill** [comment]
My `local.properties` had a comment block up top, and the script had troubles with it:

```
## This file is automatically generated by Android Studio.
# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
#
# This file must *NOT* be checked into Version Control Systems,
# as it contains information specific to your local configuration.
#
# Location of the SDK. This is only used by Gradle.
# For customization when using a Version Control System, please read the
# header note.
#Wed Sep 24 09:53:48 PDT 2014
sdk.dir=(my sdk path)
```

I deleted the comments and then the script worked fine.

File: README.md [hunk 1]
```diff
@@ -80,6 +80,19 @@ how we're organizing branches in our repository in the
 
 Say hello on our [Slack][4] channel: `#mobile`.
 
+## Alternative Build Instructions ##
+
+WordPress-Android can be compiled with [Buck][8], an alternative to Gradle,
+that makes the build process much faster. Due to actual BUck limitation, you
```
Comment (roundhill) [inline README.md]: capital `U` there ;)

File: README.md [hunk 2]
```diff
@@ -80,6 +80,19 @@ how we're organizing branches in our repository in the
...
+need to bootstrap the project by running the following command:
+
+    $ ./tools/fetch_buck_dependencies.py extlibs
+
+This command will fetch all dependencies (`.aar` and `.jar`) needed to build
+the project. Then you can run buck to build the apk:
+
+    $ buck build wpandroid
+
```
Comment (roundhill) [inline README.md]: we should add the `buck install --run wpandroid` tip here as well for running it to the device/emu

File: .buckconfig [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[alias]
+  wpandroid = //:wpandroid
+
+[android]
+  target = Google Inc.:Google APIs:21
```
Comment (roundhill) [inline .buckconfig]: Could this be bumped to the new v22 api?

**Comment by fadils** [comment]
> props = dict(line.strip().split('=') for line in open(localprops))

Do we want to parse the comments as well?

It's pretty common automatically generated `local.properties` has the following:

```
## This file is automatically generated by Android Studio.
# Do not modify this file -- YOUR CHANGES WILL BE ERASED!
#
# This file must *NOT* be checked into Version Control Systems,
# as it contains information specific to your local configuration.
#
# Location of the SDK. This is only used by Gradle.
# For customization when using a Version Control System, please read the
# header note.
#Wed Apr 15 16:14:04 WIT 2015
sdk.dir=/Users/fadilsutomo/sdk
```

If the comments are not cleaned, then the user will encounter:
                                 
```
Pr0ducer:WordPress-Android fadilsutomo$  ./tools/fetch_buck_dependencies.py extlibs
trying to read local.properties from: /Users/fadilsutomo/HackRanch/AndroidProjects/WordPress-Android/local.properties
Traceback (most recent call last):
  File "./tools/fetch_buck_dependencies.py", line 30, in <module>
    ANDROID_SDK_PATH = get_sdk_dir_from_localproperties()
  File "./tools/fetch_buck_dependencies.py", line 27, in get_sdk_dir_from_localproperties
    props = dict(line.strip().split('=') for line in open(localprops))
ValueError: dictionary update sequence element #0 has length 1; 2 is required
```

**Comment by maxme** [comment]
> My local.properties had a comment block up top,

FIxed in 471f3bb

**Comment by roundhill** [comment]
:shipit:

**Pull request merged**